api/src/gmsa_service.cpp

```diff
@@ -2151,21 +2203,28 @@ int parse_cred_spec_domainless( std::string credspec_data, creds_fetcher::krb_ti
         if (service_account_name.empty() || domain_name.empty())
             return -1;
 
-        if(contains_invalid_characters_in_credentials(domain_name) ||
-             contains_invalid_characters_in_service_name(service_account_name))
+        if(!isValidDomain(domain_name) ||
+             contains_invalid_characters_in_ad_account_name(service_account_name))
         {
             std::cout << getCurrentTime() << '\t' << "ERROR: credentialspec file is not formatted"
                                                      " properly" <<
                 std::endl;
             return -1;
         }
 
+        // get credentialspec arn
+        std::string domainless_user_arn = root["ActiveDirectoryConfig"]["HostAccountConfig"]["PluginInput"]["CredentialArn"].asString();
```

Comment:
Aren't "ActiveDirectoryConfig" and "HostAccountConfig" in the same level in the domainless json file?

Reply:
we are using correct one - https://learn.microsoft.com/en-us/virtualization/windowscontainers/manage-containers/manage-serviceaccounts

Reply:
Verified with the correct domainless json. Resolving

Reply:
Add check for errors when accessing the JSON elements